R/loading_button.R

```diff
@@ -0,0 +1,92 @@
+#' loading_button
+#'
+#' Button that becomes disabled until reset w/ `reset_loading_button`
+#'
+#' @param id the flipbox id
+#' @param label the button text (label)
+#' @param style style for button (pre-loading); character string w/ CSS styling format: "color: black; background-color: red;"
+#' @param loading_text text to show after button is clicked (e.g. during loading)
+#' @param loading_style style for button (while loading); character string w/ CSS styling format: "color: black; background-color: red;"
+#'
+#' @export
+#'
+#' @importFrom htmltools tags
+#' @family flip_box functions
+
+loading_button <- function(id, 
+                           label,
+                           style = "color: #FFFFFF; background-color: #337AB7; border-color: #2E6DA4;",
+                           loading_text = "Loading...",
+                           loading_style = "color: #FFFFFF; background-color: grey; border-color: grey;"
+) {
+  
+  htmltools::tagList(
+    tags$button(
+      id = paste0("loading_button-", id),
+      class = "btn style",
+      label
+    ),
+    tagList(
+      shiny::singleton(
+        tags$head(
+          # Used personalized kit from FA
+          tags$script(
+            src = "https://kit.fontawesome.com/fb4611ff56.js"
+          ),
+          tags$style(
+            paste0(
+              "
+              .style {",
+              style,
+              "}
+              
+              .loading_style {",
+              loading_style,
+              "}
+              "
+            )
+          ),
+          # Script for disabling button & changing text
+          tags$script(
+            HTML(
```

Comment:
move this js to a new file named "inst/srcjs/loading_button.js".  Put this js in a js function named "loading_button" and source + call this function from the html.  Also remove the `debugger`.

Reply:
eca8b3944850fc1ee9d75ecab923de7341f18380